cli/flox/src/commands/services/logs.rs

```diff
@@ -34,19 +40,35 @@ impl Logs {
         let socket = env.services_socket_path(&flox)?;
 
         let processes = ProcessStates::read(&socket)?;
-        let named_processes = super::processes_by_name_or_default_to_all(&processes, &self.names)?;
-        let names = named_processes.iter().map(|state| &state.name);
 
-        if !self.follow {
-            bail!("printing logs without following is not yet implemented");
-        }
+        if self.follow {
+            let named_processes =
+                super::processes_by_name_or_default_to_all(&processes, &self.names)?;
+            let names = named_processes.iter().map(|state| &state.name);
+            let log_stream = ProcessComposeLogStream::new(socket, names.clone(), self.tail)?;
+
+            let max_name_length = names.map(|name| name.len()).max().unwrap_or(0);
+            for log in log_stream {
+                let ProcessComposeLogLine { process, message } = log?;
+                println!("{process:<max_name_length$}: {message}",);
+            }
+        } else {
+            let [ref name] = self.names.as_slice() else {
+                bail!("When not following logs, exactly one service name must be provided");
+            };
 
-        let log_stream = ProcessComposeLogStream::new(socket, names.clone())?;
+            // Ensure the service exists
+            // Avoids attaching to a log of a non-existent service, in which case `process-compose`
+            // will block indefinitely.
+            if processes.process(name).is_none() {
+                bail!("No such service: {}", name);
```

Comment:
nit: I think we should probably standardize the wording for this, I added a `service_does_not_exist_error` helper in the `start` PR

Reply:
i copied in your changes hope that git figures that out correctly...